### PR TITLE
Add optimized library check.

### PR DESCRIPTION
**Pull request by MarkPflug**
Adds a check that verifies that all libraries are build with jit optimization enabled.

Can test against NPOI 2.5.2, which fails this check.

### CODE REVIEW

**Comment by JonDouglas** [comment]
Hey @MarkPflug,

First off, thanks so much for this contribution!

This check seems to best live in the unimplemented `pass static analysis` area. I'm fine with taking the PR as is & just letting you know that this check may be refactored later into that section. 

Or if you wanted, to just quickly refactor the `OptimizedLibraries` to go into that section. What are your thoughts? Are there other optimizations that we might check in the future to warrant it's own section?

File: Score/Models/ScoreReport.cs [hunk 1]
```diff
@@ -9,5 +9,6 @@ public class ScoreReport
         public List<ScoreSection> SupportsMultiplePlatforms { get; set; }
         public List<ScoreSection> PassStaticAnalysis { get; set; }
         public List<ScoreSection> SupportUpToDateDependencies { get; set; }
+        public List<ScoreSection> OptimizedLibraries { get; set; }
```
Comment (JonDouglas) [inline Score/Models/ScoreReport.cs:12]: Maybe this belongs in `PassStaticAnalysis`?

File: Score/Services/ScoreSectionService.cs [hunk 2]
```diff
@@ -204,5 +209,75 @@ public async Task<ScoreSection> GetUpToDateDependenciesScoreSectionAsync(Package
...
+                            Resolution = "Build with optimizations enabled, ie 'release' build."
+                        };
+                        summaries.Add(summary);
+                    }
+                }
+            }
+
+            return new ScoreSection()
+            {
+                Title = "Has optimized libraries",
```
Comment (JonDouglas) [inline Score/Services/ScoreSectionService.cs]: Should we be more specific on JIT here?

File: Score/Utilities/ScoreDumper.cs [hunk 3]
```diff
@@ -73,122 +73,146 @@ public static void DumpScore(Models.Score score)
...
-                        if (score?.ScoreReport != null)
+                    upToDateDependenciesTable
+                        .AddRow(scoreSectionTree, new Markup($"{scoreSection.CurrentScore}"),
+                            new Markup($"[bold]{scoreSection.MaxScore}[/]"));
+                }
+
+                AnsiConsole.Render(upToDateDependenciesTable);
+
+
+                var optimizedAssembliesTable = new Table();
```
Comment (JonDouglas) [inline Score/Utilities/ScoreDumper.cs]: Might be best to refactor this into the start of the `PassStaticAnalysis` section?

**Comment by MarkPflug** [comment]
Moved the jit optimizations under the Pass Static Analysis section. Took the liberty of cleaning up the table rendering a bit.

**Pull request closed without merging**